### PR TITLE
small lm training improvements

### PR DESCRIPTION
**Pull request by helpmefindaname**
fix lm training (fix https://github.com/flairNLP/flair/issues/2577 )
nice output in lm training
allow usage of lstm and gru in flair lm
remove unnecessary imports in flair/__init__.py

@alanakbik this is basically some modifications I did while training the SRU, however the SRU itself was not sucessfull enough to be merged (not only that it is a bit more complicated to install it due to special cuda requirements)

### CODE REVIEW

File: flair/models/language_model.py [hunk 1]
```diff
@@ -82,20 +83,19 @@ def forward(self, input, hidden, ordered_sequence_lengths=None):
...
 
     def init_hidden(self, bsz):
         weight = next(self.parameters()).detach()
-        return (
-            weight.new(self.nlayers, bsz, self.hidden_size).zero_().clone().detach(),
-            weight.new(self.nlayers, bsz, self.hidden_size).zero_().clone().detach(),
-        )
+        w1 = weight.new(self.nlayers, bsz, self.hidden_size).zero_().clone().detach()
+        w2 = weight.new(self.nlayers, bsz, self.hidden_size).zero_().clone().detach()
+        return w1, w2
```
Comment (alanakbik) [inline flair/models/language_model.py:98]: in a GRU, only a single state is needed

File: flair/trainers/trainer.py [hunk 2]
```diff
@@ -509,7 +510,6 @@ def train(
 
                     # depending on memory mode, embeddings are moved to CPU, GPU
                     # or deleted
```
Comment (alanakbik) [inline flair/trainers/trainer.py:512]: I guess the comments on storing embeddings should move too ;)

**Review by alanakbik — CHANGES_REQUESTED** [review]
Thanks for improving this and sorry for the late review! It mostly looks good, but the GRU is currently not working, i.e. 

```python
language_model = LanguageModel(
    dictionary,
    is_forward_lm=True,
    hidden_size=128,
    nlayers=1,
    recurrent_type="GRU",
)

# train your language model
trainer = LanguageModelTrainer(language_model, corpus)

# train your language model

trainer.train(
    "resources/LMs/test",
...
```

due to the hidden state being created for LSTM.

**Comment by helpmefindaname** [comment]
my bad, I removed the states part again, as when I removed the SRU again,
now it should work

**Review by alanakbik — APPROVED** [review]

**Comment by alanakbik** [comment]
@helpmefindaname looks great, thanks for improving this!

**Pull request merged**